test/lib/testObjectsFunctions.js

```diff
@@ -1224,6 +1224,40 @@ function register(it, expect, context) {
         expect(obj.native.existing).to.be.equal('exists');
         expect(obj.common.def).to.be.equal('Changed');
         expect(obj.native.bool).to.be.equal(false);
+
+        // now overwrite all
+        await context.adapter.setForeignObjectAsync('foreign.0.testExtendPreserve', {
+            type: 'state',
+            common: {
+                name: 'Don\'t change me',
+                type: 'string',
+                def: 'Run Forrest, Run!'
+            }, native: {
+                bool: false
+            }
+        });
+
+        options = {preserve: {common: true}};
+
+        // preserve whole common
+        await context.adapter.extendForeignObjectAsync('foreign.0.testExtendPreserve', {
+            type: 'state',
+            common: {
+                name: 'CHANGED',
+                type: 'number',
+                def: 1
+            }, native: {
+                bool: false
+            }
+        }, options);
+
+        obj = await context.adapter.getForeignObjectAsync('foreign.0.testExtendPreserve');
+
+        // preserved whole common but native is not preserved and stuff is there
+        expect(obj.common.name).to.be.equal('Don\'t change me');
+        expect(obj.common.type).to.be.equal('string');
+        expect(obj.common.def).to.be.equal('Run Forrest, Run!');
+        expect(obj.native.bool).to.be.equal(false);
```

Comment:
I think we need tests where
* the property that should be preserved is **not** present in the original object
* the property that should be preserved is **not** present in the new object
* the new object has a different type of a preserved property (e.g. number instead of object)

Reply:
added it as parts of the preserve tests, wouldn't split it up in separate tests